File: frontend/src/types/validation.ts
export interface ValidationResult {
  id: number;
  target_date: string;
  supplier_code: string;
  target_key: string;
  predicted_qty: number;
  actual_qty: number;
  accuracy: number;
  usage_rate: number;
  unwrap_rate: number;
  is_overpredict: boolean;
  is_underpredict: boolean;
}

export interface ReportSummary {
  period_start: string;
  period_end: string;
  supplier_code: string | null;
  total_predictions: number;
  avg_accuracy: number;
  avg_usage_rate: number;
  avg_unwrap_rate: number;
  overpredict_count: number;
  underpredict_count: number;
}

File: backend/schemas/validation.py
from datetime import date
from pydantic import BaseModel


class ValidationResultOut(BaseModel):
    id: int
    target_date: date
    supplier_code: str
    target_key: str
    predicted_qty: int
    actual_qty: int
    accuracy: float
    usage_rate: float
    unwrap_rate: float
    is_overpredict: bool
    is_underpredict: bool

    model_config = {"from_attributes": True}


class ValidationRunRequest(BaseModel):
    target_date: date
    supplier_code: str | None = None


class ReportSummary(BaseModel):
    period_start: date
    period_end: date
    supplier_code: str | None
    total_predictions: int
    avg_accuracy: float
    avg_usage_rate: float
    avg_unwrap_rate: float
    overpredict_count: int
    underpredict_count: int
